### PR TITLE
Add dotenv backend (#3)

### LINKED ISSUE #3: Add backend for .env file
Add backend that uses https://github.com/jpadilla/django-dotenv (MIT) to parse a file

Packaging note: define an extra to make the dependency optional

### PR DESCRIPTION
**Pull request by merwok**
Fixes #3 

To be merged after CI PR and API change PR.

### CODE REVIEW

File: configstore/backends/dotenv.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class DotenvBackend(object):
+
+    def __init__(self, dotenv_path):
+        if dotenv is None:
+            raise ImportError('install configstore[dotenv] to use '
+                              'the dotenv backend')
+
+        with open(dotenv_path) as file:
```
Comment (merwok) [inline configstore/backends/dotenv.py]: The docker_secret backend could also check at init time that the directory exists.  Just saying.

**Comment by crevetor** [comment]
This contains a bunch of unrelated changes. Could you split it up into different PRs ?

**Comment by merwok** [comment]
It appears this way because I branched off another PR, not master.  It will look right after I merge master.

File: configstore/backends/dotenv.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+except ImportError:  # pragma: no cover
+    dotenv = None
+
+
+class DotenvBackend(object):
+
+    def __init__(self, dotenv_path):
+        if dotenv is None:
+            raise ImportError('install configstore[dotenv] to use '
+                              'the dotenv backend')
```
[the remark this answers is not in the document]
Comment (crevetor) [reply configstore/backends/dotenv.py]: Why not just let the initial importError bubble up ?

[the remark this answers is not in the document]
Comment (merwok) [reply configstore/backends/dotenv.py]: Then we can’t have DockerBackend imported in `__init__` modules, and in settings files people would have to put the import inside a conditional (local vs. server envs).  With the idiom «import always works / instantiating may raise», people can import configstore at the top like any module and only contain the instantiation in an if.

[the remark this answers is not in the document]
Comment (crevetor) [reply configstore/backends/dotenv.py]: Sounds legit. Let's leave the ImportError for now. We might revisit at some point.

**Pull request merged**